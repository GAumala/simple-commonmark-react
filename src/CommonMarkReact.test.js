const React = require('react')
const ReactDOMServer = require('react-dom/server')

const { renderNodes} = require('./CommonMarkReact.js')

const renderToString = (markdown, options) => {
  const nodes = renderNodes(markdown, Object.assign({className: 'markdown'}, options))
  const rootElement = React.createElement('div', null, nodes)
  return ReactDOMServer.renderToStaticMarkup(rootElement)
}

test('Renders markdown tags: header and paragraph with custom class names', () => {
  const markdown = '## Title\n\nHello World!\n'
  const expectedHtml = '<div><h2 class="markdown">Title</h2><p class="markdown">Hello World!</p></div>'
  const renderedString = renderToString(markdown)
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders inline markdown tags: bold, code, softbreak and emph with custom class names', () => {
  const markdown = 'Hello World!\n*this* is italics, **this** is bold and `this` is code'
  const expectedHtml = '<div><p class="markdown">Hello World!\n<em class="markdown">this</em> is italics, <strong class="markdown">this</strong> is bold and <code class="markdown">this</code> is code</p></div>'
  const renderedString = renderToString(markdown)
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders line breaks as <br/> if allowSoftBreaks option is active', () => {
  const markdown = 'Hello World!\n*this* is italics, **this** is bold and `this` is code'
  const expectedHtml = '<div><p class="markdown">Hello World!<br/><em class="markdown">this</em> is italics, <strong class="markdown">this</strong> is bold and <code class="markdown">this</code> is code</p></div>'
  const renderedString = renderToString(markdown, {allowSoftBreaks: true})
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders markdown tags: header, paragraph and code_block with custom class names', () => {
  const markdown = '# Code Test\n\n Check this code\n\n```c\n\n#include <stdio.h>\n```'
  const renderedString = renderToString(markdown)
  const expectedHtml = '<div><h1 class="markdown">Code Test</h1><p class="markdown">Check this code</p><pre class="language-c markdown"><code class="markdown">\n#include &lt;stdio.h&gt;\n</code></pre></div>'
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders code_block without language name with custom class names', () => {
  const markdown = '# Code Test\n\n Check this code\n\n```\nSome pseudo code\n```'
  const renderedString = renderToString(markdown)
  const expectedHtml = '<div><h1 class="markdown">Code Test</h1><p class="markdown">Check this code</p><pre class="markdown"><code class="markdown">Some pseudo code\n</code></pre></div>'
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders markdown link with custom class names', () => {
  const markdown = 'Check out this [link](https://fsf.org) and [this one](/local/path "some server url") with title'
  const renderedString = renderToString(markdown)
  const expectedHtml = '<div><p class="markdown">Check out this <a class="markdown" href="https://fsf.org">link</a> and <a class="markdown" title="some server url" href="/local/path">this one</a> with title</p></div>'
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders markdown images with custom class names', () => {
  const markdown = 'Check out this ![image](https://somewebsite.org/pic.png) it is inline\n\nAlso this one:\n\n![local pic](/public/photo.png "my pic")'
  const renderedString = renderToString(markdown)
  const expectedHtml = '<div><p class="markdown">Check out this <img class="markdown" alt="image" src="https://somewebsite.org/pic.png"/> it is inline</p><p class="markdown">Also this one:</p><p class="markdown"><img class="markdown" title="my pic" alt="local pic" src="/public/photo.png"/></p></div>'
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders bullet lists with custom class names', () => {
  const markdown = '- item with [link](https://eff.org)\n- item with `code`\n- item with **bold**\n- item with *emph*\n'
  const renderedString = renderToString(markdown)
  const expectedHtml = '<div><ul class="markdown"><li class="markdown">item with <a class="markdown" href="https://eff.org">link</a></li><li class="markdown">item with <code class="markdown">code</code></li><li class="markdown">item with <strong class="markdown">bold</strong></li><li class="markdown">item with <em class="markdown">emph</em></li></ul></div>'
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders ordered lists with custom class names', () => {
  const markdown = '1) item with [link](https://eff.org)\n2) item with `code`\n3) item with **bold**\n4) item with *emph*\n'
  const renderedString = renderToString(markdown)
  const expectedHtml = '<div><ol class="markdown"><li class="markdown">item with <a class="markdown" href="https://eff.org">link</a></li><li class="markdown">item with <code class="markdown">code</code></li><li class="markdown">item with <strong class="markdown">bold</strong></li><li class="markdown">item with <em class="markdown">emph</em></li></ol></div>'
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders loose lists with custom class names', () => {
  const markdown = '- item with [link](https://eff.org)\n\n- item with *emph*\n\n'
  const renderedString = renderToString(markdown)
  const expectedHtml = '<div><ul class="markdown"><li class="markdown"><p class="markdown">item with <a class="markdown" href="https://eff.org">link</a></p></li><li class="markdown">item with <em class="markdown">emph</em></li></ul></div>'
  expect(renderedString).toEqual(expectedHtml)
})

test('Renders correctly when customProps is undefined', () => {
  const markdown = '# Code Test\n\n Check this code\n\n```c\n\n#include <stdio.h>\n```'
  const nodes = renderNodes(markdown)
  const rootElement = React.createElement('div', null, nodes)
  const renderedString =  ReactDOMServer.renderToStaticMarkup(rootElement)
  const expectedHtml = '<div><h1>Code Test</h1><p>Check this code</p><pre class="language-c "><code>\n#include &lt;stdio.h&gt;\n</code></pre></div>'
  expect(renderedString).toEqual(expectedHtml)
})
