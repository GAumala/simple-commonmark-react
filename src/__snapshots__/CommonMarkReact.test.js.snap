// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders bullet lists with custom class names 1`] = `
<div>
  <ul
    className="markdown"
  >
    <li
      className="markdown"
    >
      item with 
      <a
        className="markdown"
        href="https://eff.org"
      >
        link
      </a>
    </li>
    <li
      className="markdown"
    >
      item with 
      <code
        className="markdown"
      >
        code
      </code>
    </li>
    <li
      className="markdown"
    >
      item with 
      <strong
        className="markdown"
      >
        bold
      </strong>
    </li>
    <li
      className="markdown"
    >
      item with 
      <em
        className="markdown"
      >
        emph
      </em>
    </li>
  </ul>
</div>
`;

exports[`Renders code_block without language name with custom class names 1`] = `
<div>
  <h1
    className="markdown"
  >
    Code Test
  </h1>
  <p
    className="markdown"
  >
    Check this code
  </p>
  <pre
    className="markdown"
  >
    <code
      className="markdown"
    >
      Some pseudo code
      
    </code>
  </pre>
</div>
`;

exports[`Renders correctly when customProps is undefined 1`] = `
<div>
  <h1
    className={undefined}
  >
    Code Test
  </h1>
  <p
    className={undefined}
  >
    Check this code
  </p>
  <pre
    className="language-c "
  >
    <code
      className={undefined}
    >
      
      #include &lt;stdio.h&gt;
      
    </code>
  </pre>
</div>
`;

exports[`Renders header and html block 1`] = `
<div>
  <h1
    className="markdown"
  >
    Html Block
  </h1>
  <div
    className="markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p style=\\"text-align:right\\">Aligned to right.</p>",
      }
    }
  />
</div>
`;

exports[`Renders inline markdown tags: bold, code, softbreak and emph with custom class names 1`] = `
<div>
  <p
    className="markdown"
  >
    Hello World
    !
    
    
    <em
      className="markdown"
    >
      this
    </em>
     is italics, 
    <strong
      className="markdown"
    >
      this
    </strong>
     is bold and 
    <code
      className="markdown"
    >
      this
    </code>
     is code
  </p>
</div>
`;

exports[`Renders line breaks as <br/> if allowSoftBreaks option is active 1`] = `
<div>
  <p
    className="markdown"
  >
    Hello World
    !
    <br
      className="markdown"
    />
    <em
      className="markdown"
    >
      this
    </em>
     is italics, 
    <strong
      className="markdown"
    >
      this
    </strong>
     is bold and 
    <code
      className="markdown"
    >
      this
    </code>
     is code
  </p>
</div>
`;

exports[`Renders loose lists with custom class names 1`] = `
<div>
  <ul
    className="markdown"
  >
    <li
      className="markdown"
    >
      <p
        className="markdown"
      >
        item with 
        <a
          className="markdown"
          href="https://eff.org"
        >
          link
        </a>
      </p>
    </li>
    <li
      className="markdown"
    >
      item with 
      <em
        className="markdown"
      >
        emph
      </em>
    </li>
  </ul>
</div>
`;

exports[`Renders markdown block quote with bold and italics 1`] = `
<div>
  <blockquote
    className="markdown"
  >
    <p
      className="markdown"
    >
      This is some quote with 
      <strong
        className="markdown"
      >
        bold
      </strong>
       and 
      <em
        className="markdown"
      >
        italics
      </em>
      .
    </p>
  </blockquote>
</div>
`;

exports[`Renders markdown images with custom class names 1`] = `
<div>
  <p
    className="markdown"
  >
    Check out this 
    <img
      alt="image"
      className="markdown"
      src="https://somewebsite.org/pic.png"
    />
    
     it is inline
  </p>
  <p
    className="markdown"
  >
    Also this one:
  </p>
  <p
    className="markdown"
  >
    <img
      alt="local pic"
      className="markdown"
      src="/public/photo.png"
      title="my pic"
    />
    
  </p>
</div>
`;

exports[`Renders markdown link with custom class names 1`] = `
<div>
  <p
    className="markdown"
  >
    Check out this 
    <a
      className="markdown"
      href="https://fsf.org"
    >
      link
    </a>
     and 
    <a
      className="markdown"
      href="/local/path"
      title="some server url"
    >
      this one
    </a>
     with title
  </p>
</div>
`;

exports[`Renders markdown tags: header and paragraph with custom class names 1`] = `
<div>
  <h2
    className="markdown"
  >
    Title
  </h2>
  <p
    className="markdown"
  >
    Hello World
    !
  </p>
</div>
`;

exports[`Renders markdown tags: header, paragraph and code_block with custom class names 1`] = `
<div>
  <h1
    className="markdown"
  >
    Code Test
  </h1>
  <p
    className="markdown"
  >
    Check this code
  </p>
  <pre
    className="language-c markdown"
  >
    <code
      className="markdown"
    >
      
      #include &lt;stdio.h&gt;
      
    </code>
  </pre>
</div>
`;

exports[`Renders ordered lists starting from arbitrary index with custom class names 1`] = `
<div>
  <ol
    className="markdown"
    start={4}
  >
    <li
      className="markdown"
    >
      item with 
      <strong
        className="markdown"
      >
        bold
      </strong>
    </li>
    <li
      className="markdown"
    >
      item with 
      <code
        className="markdown"
      >
        code
      </code>
    </li>
  </ol>
</div>
`;

exports[`Renders ordered lists with custom class names 1`] = `
<div>
  <ol
    className="markdown"
  >
    <li
      className="markdown"
    >
      item with 
      <a
        className="markdown"
        href="https://eff.org"
      >
        link
      </a>
    </li>
    <li
      className="markdown"
    >
      item with 
      <code
        className="markdown"
      >
        code
      </code>
    </li>
    <li
      className="markdown"
    >
      item with 
      <strong
        className="markdown"
      >
        bold
      </strong>
    </li>
    <li
      className="markdown"
    >
      item with 
      <em
        className="markdown"
      >
        emph
      </em>
    </li>
  </ol>
</div>
`;

exports[`Renders sections separated by thematic breaks with custom class names 1`] = `
<div>
  <h1
    className="markdown"
  >
    Section 1
  </h1>
  <p
    className="markdown"
  >
    Some text.
  </p>
  <hr
    className="markdown"
  />
  <h1
    className="markdown"
  >
    Section 2
  </h1>
  <p
    className="markdown"
  >
    Some more text.
  </p>
</div>
`;
